discovery.go

```diff
@@ -29,7 +64,10 @@ func init() {
 
 func main() {
 	log.SetFlags(0)
-	handling.Setup()
+	viper.Set("etcd", mustHostOnlyURL(viper.GetString("etcd")))
+	viper.Set("host", mustHostOnlyURL(viper.GetString("host")))
+
+	handling.Setup(viper.GetString("etcd"))
```

Comment:
I'd rather set the viper var to the 100% valid url returned by mustHostOnlyURL, just so that the viper config contains a authoritative, correct copy of all config options. Thoughts?